From e2ea6bd59a5ee87efd74f5ffbed39ca19e48f677 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:20:18 -0400
Subject: [fs] sunrpc: remove __rcu annotation from struct gss_cl_ctx->gc_gss_ctx

Message-id: <1411050057-12417-56-git-send-email-steved@redhat.com>
Patchwork-id: 96200
O-Subject: [RHEL7.1 PATCH 55/94] sunrpc: remove __rcu annotation from struct gss_cl_ctx->gc_gss_ctx
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Commit 5b22216e11f7 (nfs: __rcu annotations) added a __rcu annotation to
the gc_gss_ctx field. I see no rationale for adding that though, as that
field does not seem to be managed via RCU at all.

Cc: Arnd Bergmann <arnd@arndb.de>
Cc: Paul McKenney <paulmck@linux.vnet.ibm.com>
Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: a3b255717fed1cad0dd4ed5be77114d32ef22a6d
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/sunrpc/auth_gss.h b/include/linux/sunrpc/auth_gss.h
index cbc6875..36eebc4 100644
--- a/include/linux/sunrpc/auth_gss.h
+++ b/include/linux/sunrpc/auth_gss.h
@@ -69,7 +69,7 @@ struct gss_cl_ctx {
  enum rpc_gss_proc gc_proc;
  u32   gc_seq;
  spinlock_t  gc_seq_lock;
- struct gss_ctx __rcu *gc_gss_ctx;
+ struct gss_ctx  *gc_gss_ctx;
  struct xdr_netobj gc_wire_ctx;
  struct xdr_netobj gc_acceptor;
  u32   gc_win;
-- 
1.7.1